src/containers/oracle.js

```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
-                break;
-              }
-              default: {
-                console.warn('Invalid oracle type');
-                oracle = undefined;
-                break;
-              }
-            }
-            break;
-          }
-          default: {
-            console.warn('Oracle exists but cant determine status.');
-            oracle = undefined;
-            break;
-          }
+    const oracle = _.find(getOraclesSuccess, { address: this.state.address });
+
+    if (oracle) {
+      const { token, status, endBlock } = oracle;
+      let config;
+
+      /** Determine what config to use in current card * */
+      if (status === 'VOTING' && token === QTUM) {
+        config = {
+          name: 'BETTING',
+          breadcrumbLabel: 'Betting',
+          cardInfo: {
+            steps: {
+              current: 1,
+              value: [{
+                title: 'Topic created',
+                description: `Block No. ${oracle.blockNum}`,
+              },
+              {
+                title: 'Betting',
+                description: `Block No. ${oracle.blockNum + 1} - ${oracle.endBlock}`,
+              },
+              {
+                title: 'Result Setting',
+                description: `Block No. ${oracle.endBlock + 1} - ${oracle.resultSetEndBlock}`,
+              },
+              ],
+            },
+            messages: [
+            ],
+          },
+          cardAction: {
+            skipToggle: false,
+            beforeToggle: {
+              btnText: 'Bet',
+            },
+            afterToggle: {
+              showAmountInput: true,
+              btnText: 'Confirm',
+            },
+          },
+        };
+      } else if (status === 'WAITRESULT' && token === QTUM) {
+        config = {
+          name: 'SETTING',
+          breadcrumbLabel: 'Setting',
+          cardInfo: {
+            steps: {
+              current: 2,
+              value: [{
+                title: 'Topic created',
+                description: `Block No. ${oracle.blockNum}`,
+              },
+              {
+                title: 'Betting',
+                description: `Block No. ${oracle.blockNum + 1} - ${oracle.endBlock}`,
+              },
+              {
+                title: 'Result Setting',
+                description: `Block No. ${oracle.endBlock + 1} - ${oracle.resultSetEndBlock}`,
+              },
+              ],
+            },
+            messages: [
+              {
+                text: `Result setter ${oracle.resultSetterAddress}`,
+                type: 'default',
+              },
+              {
+                text: `Consensus Threshold ${oracle.consensusThreshold || 0}. This value indicates the amount of BOT needed for result setting.`,
+                type: 'default',
+              },
+
+              {
+                text: 'Upon clicking Confirm, you will need to wait for BOT token to get approved. Those BOT amount will automatically be used to set result afterwards.',
```

Comment:
Should we also say `Please don't leave this screen`?

Reply:
Sure, I'm going to add that text to above button which could be more obvious.

Reply:
good point. Added above text